numcosmo/perturbations/nc_hipert_adiab.c

```diff
@@ -157,113 +165,84 @@ _nc_hipert_adiab_finalize (GObject *object)
-static void _nc_hipert_adiab_eval_system (NcmHOAA *hoaa, NcmModel *model, const gdouble t, const gdouble k, gdouble *nu, gdouble *dlnmnu, gdouble *Vnu);
-
-static guint _nc_hipert_adiab_nsing (NcmHOAA *hoaa, NcmModel *model, const gdouble k);
-static void _nc_hipert_adiab_get_sing_info (NcmHOAA *hoaa, NcmModel *model, const gdouble k, const guint sing, gdouble *ts, gdouble *dts_i, gdouble *dts_f, NcmHOAASingType *st);
-static gdouble _nc_hipert_adiab_eval_sing_mnu (NcmHOAA *hoaa, NcmModel *model, const gdouble t_m_ts, const gdouble k, const guint sing);
-static gdouble _nc_hipert_adiab_eval_sing_dlnmnu (NcmHOAA *hoaa, NcmModel *model, const gdouble t_m_ts, const gdouble k, const guint sing);
-static void _nc_hipert_adiab_eval_sing_system (NcmHOAA *hoaa, NcmModel *model, const gdouble t_m_ts, const gdouble k, const guint sing, gdouble *nu, gdouble *dlnmnu, gdouble *Vnu);
-
-static gdouble _nc_hipert_adiab_eval_powspec_factor (NcmHOAA *hoaa, NcmModel *model);
-
-static void _nc_hipert_adiab_prepare (NcmHOAA *hoaa, NcmModel *model);
+static gdouble _nc_hipert_adiab_eval_m            (NcmCSQ1D *csq1d, NcmModel *model, const gdouble t, const gdouble k);
+static gdouble _nc_hipert_adiab_eval_mnu            (NcmCSQ1D *csq1d, NcmModel *model, const gdouble t, const gdouble k);
+static gdouble _nc_hipert_adiab_eval_nu             (NcmCSQ1D *csq1d, NcmModel *model, const gdouble t, const gdouble k);
+static gdouble _nc_hipert_adiab_eval_xi             (NcmCSQ1D *csq1d, NcmModel *model, const gdouble t, const gdouble k);
+static gdouble _nc_hipert_adiab_eval_F1             (NcmCSQ1D *csq1d, NcmModel *model, const gdouble t, const gdouble k);
+static gdouble _nc_hipert_adiab_eval_F2             (NcmCSQ1D *csq1d, NcmModel *model, const gdouble t, const gdouble k);
+static void _nc_hipert_adiab_eval_system            (NcmCSQ1D *csq1d, NcmModel *model, const gdouble t, const gdouble k, gdouble *nu, gdouble *dlnmnu, gdouble *Vnu);
+static gdouble _nc_hipert_adiab_eval_powspec_factor (NcmCSQ1D *csq1d, NcmModel *model);
 
 static void
 nc_hipert_adiab_class_init (NcHIPertAdiabClass *klass)
 {
   GObjectClass *object_class = G_OBJECT_CLASS (klass);
-  NcmHOAAClass *hoaa_class   = NCM_HOAA_CLASS (klass);
+  NcmCSQ1DClass *csq1d_class   = NCM_CSQ1D_CLASS (klass);
 
   object_class->set_property = &_nc_hipert_adiab_set_property;
   object_class->get_property = &_nc_hipert_adiab_get_property;
   object_class->dispose      = &_nc_hipert_adiab_dispose;
   object_class->finalize     = &_nc_hipert_adiab_finalize;
+  
+  csq1d_class->eval_m       = &_nc_hipert_adiab_eval_m;
+  csq1d_class->eval_mnu    = &_nc_hipert_adiab_eval_mnu;
+  csq1d_class->eval_nu     = &_nc_hipert_adiab_eval_nu;
+  csq1d_class->eval_xi     = &_nc_hipert_adiab_eval_xi;
+  csq1d_class->eval_F1     = &_nc_hipert_adiab_eval_F1;
+  csq1d_class->eval_F2     = &_nc_hipert_adiab_eval_F2;
+  csq1d_class->eval_V      = NULL;
```

Comment:
Remove V (or use as a placeholder for other methods).

Reply:
Removed from adiab object and csq1d